lab_1_keywords_tfidf/main.py

```diff
@@ -3,103 +3,157 @@
 Extract keywords based on frequency related metrics
 """
 from typing import Optional, Union
+import math
+
+
+def check_input_type(check_arg, check_type, check_token=None, check_value=None, can_be_empty=True):
+    if not isinstance(check_arg, check_type):
+        return False
+    if not check_arg and not can_be_empty:
+        return False
+    if check_type == list:
+        for i in check_arg:
+            if not isinstance(i, check_token):
+                return False
+    if check_type == dict:
+        for i in check_arg:
+            if not isinstance(i, check_value):
+                return False
+    return True
+
+
+def check_num(num_arg):
+    if not isinstance(num_arg, int) or type(num_arg) == bool:
+        return False
+    if num_arg < 0:
+        return False
+    return True
 
 
 def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
     Removes punctuation, casts to lowercase, splits into tokens
-
```

Comment:
please restore empty lines in docstrings: students are not supposed to change anything relating to function signature